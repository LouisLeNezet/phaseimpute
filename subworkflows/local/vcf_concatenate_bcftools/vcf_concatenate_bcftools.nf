include { BCFTOOLS_CONCAT } from '../../../modules/nf-core/bcftools/concat/main'
include { BCFTOOLS_INDEX  } from '../../../modules/nf-core/bcftools/index/main'

workflow VCF_CONCATENATE_BCFTOOLS {

    take:
    ch_vcf_tbi                            // channel: [ val(meta), vcf, tbi ]

    main:

    // Remove chromosome from meta
    ch_vcf_tbi_grouped = ch_vcf_tbi.map{ meta, vcf, tbi ->
                        return [['id' : meta.id], vcf, tbi]
                        }
    // Group by ID
    ch_vcf_tbi_grouped = ch_vcf_tbi_grouped.groupTuple( by:[0] )

    // Ligate and concatenate chunks
    BCFTOOLS_CONCAT(ch_vcf_tbi_grouped)

    // Index concatenated VCF
    BCFTOOLS_INDEX(BCFTOOLS_CONCAT.out.vcf)

    // Join VCFs and TBIs
    ch_imputed_vcf_tbi = BCFTOOLS_CONCAT.out.vcf.join(BCFTOOLS_INDEX.out.tbi)

    emit:
    ch_imputed_vcf_tbi                          // channel:  [ meta, vcf, tbi ]

    }
