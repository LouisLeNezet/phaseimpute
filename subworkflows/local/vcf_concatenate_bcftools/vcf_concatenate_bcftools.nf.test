nextflow_workflow {

    name "Test Subworkflow VCF_CONCATENATE_BCFTOOLS"
    script "./vcf_concatenate_bcftools.nf"
    workflow "VCF_CONCATENATE_BCFTOOLS"


    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/vcf_concatenate_bcftools"
    tag "vcf_concatenate_bcftools"

    tag "bcftools"
    tag "bcftools/concat"
    tag "bcftools/index"

    test("Should run without failures") {
    config "./tests/nextflow.config"
        when {
            workflow {
                """
                // define inputs of the workflow here. Example:
                // input[0] = file("test-file.txt")

                input[0] = Channel.fromList([
                    [
                        [id: "chr21"],
                        file("https://github.com/atrigila/test-datasets/raw/phaseimpute/data/quilt/quilt.chr21.16570070.16609998.vcf.gz",checkIfExist:true)
                    ],
                    [
                        [id: "chr22"],
                        file("https://github.com/atrigila/test-datasets/raw/phaseimpute/data/quilt/quilt.chr22.16570065.16609999.vcf.gz",checkIfExist:true)
                    ]
                ])


                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
