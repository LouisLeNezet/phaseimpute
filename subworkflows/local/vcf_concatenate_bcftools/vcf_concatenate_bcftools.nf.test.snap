{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr21"
                        },
                        "chr21.vcf.gz:md5,c657d32f8b9ac60eb0f2a356fd14a271"
                    ],
                    [
                        {
                            "id": "chr22"
                        },
                        "chr22.vcf.gz:md5,df689c605e8ee1adfce32ad176067a8d"
                    ]
                ],
                "ch_concat_vcf": [
                    [
                        {
                            "id": "chr21"
                        },
                        "chr21.vcf.gz:md5,c657d32f8b9ac60eb0f2a356fd14a271"
                    ],
                    [
                        {
                            "id": "chr22"
                        },
                        "chr22.vcf.gz:md5,df689c605e8ee1adfce32ad176067a8d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-11T16:37:31.573470439"
    }
}